weaviate/collection/grpc.py

```diff
@@ -376,13 +383,39 @@ def near_audio(
         self._autocut = autocut
         self._filters = filters
 
+        self._metadata = return_metadata
+        if return_properties is not None:
+            self._default_props = self._default_props.union(return_properties)
+        return cast(List[SearchResult], self.__call().results)
+
+    def generative(
+        self,
+        single: Optional[str] = None,
+        grouped: Optional[str] = None,
+        grouped_properties: Optional[List[str]] = None,
+        auto_limit: Optional[int] = None,
+        filters: Optional[_Filters] = None,
+        return_metadata: Optional[MetadataQuery] = None,
+        return_properties: Optional[PROPERTIES] = None,
+    ) -> SearchResponse:
+        if single is None and grouped is None:
+            raise ValueError(
+                "Either single_response or grouped response must be not None for generative search."
+            )
+
+        self._generative_single = single
+        self._generative_grouped = grouped
+        self._generative_grouped_properties = grouped_properties
+        self._autocut = auto_limit
+        self._filters = filters
+
         self._metadata = return_metadata
```

Comment:
Further to my point above, should `MetadataQuery` here have `generative=True` hard-populated `if single is not None`?

Reply:
not needed (also not possible, there is no generative in MetadataQuery). The server detects if it is a generative query and includes it in the response automatically

Reply:
got confused with `_MetadataReturn` 😅